hw/ip/edn/doc/_index.md

```diff
@@ -0,0 +1,270 @@
+On exiting, the EDN issues an `uninstantiate` command to destroy the associated CSRNG instance.
+To simplify the boot process, this mode is enabled as the default mode on reset.
+Therefore if no other registers are set after reset, a single write to the `EDN_ENABLE` bit will start the flow of entropy to the endpoints in boot-time request mode.
+
+Once firmware initialization is complete, it is important to exit this mode if the endpoints ever need FIPS-approved random values.
+This is done by either *clearing* the `EDN_ENABLE` bit or *setting* the `BOOT_REQ_DIS` bit in {{< regref "CTRL" >}} to halt the boot-time request state machine.
+Firmware must then wait for successful the shutdown of the state machine by polling the `REQ_MODE_SM_STS` field of the {{< regref "SUM_STS" >}} register.
+
+#### Note on Security Considerations when Using Boot-time Request Mode
+
+Boot-time request mode is not intended for normal operation, as it tolerates the potential use of preliminary seeds for the attached CSRNG instance.
+These preliminary seeds are described as "pre-FIPS" since they are released from the `entropy_src` before the complete start-up health-checks recommended by FIPS have been completed.
+Thus pre-FIPS seeds have weaker guarantees on the amount of physical entropy included in their creation.
+As detailed in the [`entropy_src` documentation]({{< relref "hw/ip/entropy_src/doc" >}}), only the first CSRNG seed created after reset is pre-FIPS, all following seeds from the `entropy_src` are passed through the full FIPS-approved health checks.
+Therefore at most one EDN can receive a pre-FIPS seed after reset.
+Since boot-time request mode EDN streams may be FIPS non-compliant, firmware must at some point disable boot-time request mode and reinitialize the EDN for either firmware-driven operation or auto request mode.
+
+#### Multiple EDNs in Boot-time Request Mode
+
+If many endpoints require boot-time entropy (and if broadcast entropy is deemed unacceptable for some peripherals) multiple boot-time EDNs may be required, as the EDN has a fixed maximum number of peripheral ports.
+Since physical entropy generation takes time, there exists a mechanism to prioritize the EDNs, to match the boot priority of each group of attached endpoints.
+To establish an order to the instantiation of each EDN, enable them one at a time.
+To ensure that the most recently enabled EDN will get next priority for physical entropy, poll the `BOOT_INST_ACK` field in the {{< regref "SUM_STS" >}} register before enabling the following EDN.
+
+The first EDN to enter boot-time request mode will receive random values generated from the initial pre-FIPS CSRNG instance roughly 2ms after reset.
+As the `entropy_src` only creates one pre-FIPS seed, any other boot-time request mode EDNs will actually receive values from FIPS-approved instances, with an approximate delay of 5ms between each instantiation.
```

Comment:
This is not quite right.  However I can perhaps alleviate your confusion by pointing out some semantic overloading that has happened here between NIST and SystemVerilog. 

There is only _one_ entropy_src, and only _one_ CSRNG SystemVerilog _module_ instance.  However NIST talks about a DRBG as having multiple "instances" internally, which basically from a hardware point of view mean separate state vectors.  So in the NIST sense the word instance is very symbolic.  (It makes a lot more sense when you consider software implementations, and think of an "instance" as just some tiny data structure).

So now in our case, each port has an "instance" (in the NIST sense) of the state vector behind it, not a whole CSRNG instance (in the SystemVerilog sense).

So the CSRNG collects a sequence of seeds from the entopy_src, and collects the seeds, and each EDN gets a unique seed.

Any recommendations on how to draw the diagram in this case?

Reply:
Diagram aside, the point of this paragraph is not the topology, it's the timing.   That said I've rewritten this paragraph for greater emphasis:
```
If using boot-time request mode, the CSRNG seed.material used for the first-activated EDN is the special pre-FIPS seed, which is specifically tested quickly to improve latency.
The first random values distributed from this EDN will therefore be available roughly 2ms after reset.
The `entropy_src` only creates only one pre-FIPS seed, so any other EDNs must wait for their seeds to pass the full FIPS-recommended health checks.
This means that each subsequent EDN must wait an additional 5ms before it can start distributing data.
For instance, if there are three boot-time request mode EDN's in the system, the first will start distributing data 2ms after reset, the second will start distributing data 7ms after reset, and the third will start distributing data 12ms after reset.
```

Reply:
Okay.   Redrawing this figure to add the entropy source, is actually harder than it should be. That said, rather than adding the entropy source to the figure, I've added some clarifying text to address your request.
```
In general, the OpenTitan random number subsystem consists of one `entropy_src`, one CSRNG, and one or more EDNs.
The `entropy_src` only supports one connection to a CSRNG, but the CSRNG has multiple application interface ports for connecting to EDN's or other hardware blocks.
The diagram below shows an example topology where two EDN modules are used to distribute `genbits` from the CSRNG to peripheral modules.
The `entropy_src` is not shown in this figure.
```

Reply:
Actually, never mind my whinging.  The edn_top_diag.vsdx is far simpler than some of Marks other drawings.  That wasn't so hard at all.  The diagram you are looking for will be in the next commit.

Reply:
Ah, thank you! This makes much more sense to me now!